mentorup/src/components/Dashboard/Dashboard.js

```diff
@@ -14,30 +14,37 @@ import {
 
 const Dashboard = () => {
   const { data: user } = useUser();
+  //If user is undefined, uses the local storage
+  const currentUserInfo = user ? user : JSON.parse(localStorage.getItem("currentUserInfo"));
   const firestore = useFirestore();
-
-  //userUID should not be a magic number -> get data from firebase
-  const userUID = "G77IDapZAJU5z4vZHwf4pY3fLH12";
+  const userRef = firestore
```

Comment:
I think this should be done asynchronously in the background so that it does not block the main thread. You need to do it in the initial useEffect() method that React provides to you. The only things you should be defining at the parent level are states. Load data into the states via async methods. Not directly like this.